tasks/run_puppet_device.rb

```diff
@@ -42,8 +42,7 @@ def read_device_certificate_fingerprints(cert_name)
   return nil unless certificate
   fingerprints = {}
   fingerprints['default'] = certificate.fingerprint
-  ssl_host = Puppet::SSL::Host.new
-  mdas = ssl_host.suitable_message_digest_algorithms
+  mdas = [:SHA1, :SHA224, :SHA256, :SHA384, :SHA512] # ssl_host.suitable_message_digest_algorithms was removed in Puppet 6, specifying the array directly
```

Comment:
@da-ar 

If we are interested in a 'canonical' list, `suitable_message_digest_algorithms` appears to have moved to `default_digest_algorithm` in Puppet 6.0; and is FIPS-aware, as per PUP-8021:

https://github.com/puppetlabs/puppet/blob/6.0.x/lib/puppet/defaults.rb

```
mdas  = Puppet.valid_digest_algorithms
```

Defaults in Puppet 6.0 also offers `Puppet.default_digest_algorithm` which may be enough to satisfy FM-7115.

Reply:
this is being addressed in https://github.com/puppetlabs/device_manager/pull/29